lib/load.php

```diff
@@ -24,31 +24,36 @@
 require dirname( __FILE__ ) . '/register.php';
 require dirname( __FILE__ ) . '/demo.php';
 
-// Register server-side code for individual blocks.
-if ( ! function_exists( 'render_block_core_archives' ) ) {
-	require dirname( __FILE__ ) . '/../packages/block-library/src/archives/index.php';
-}
-if ( ! function_exists( 'render_block_core_block' ) ) {
-	require dirname( __FILE__ ) . '/../packages/block-library/src/block/index.php';
-}
-if ( ! function_exists( 'render_block_core_categories' ) ) {
-	require dirname( __FILE__ ) . '/../packages/block-library/src/categories/index.php';
-}
-// Currently merged in core as `gutenberg_render_block_core_latest_comments`,
-// expected to change soon.
-if ( ! function_exists( 'render_block_core_latest_comments' )
-	&& ! function_exists( 'gutenberg_render_block_core_latest_comments' ) ) {
-	require dirname( __FILE__ ) . '/../packages/block-library/src/latest-comments/index.php';
-}
-if ( ! function_exists( 'render_block_core_latest_posts' ) ) {
-	require dirname( __FILE__ ) . '/../packages/block-library/src/latest-posts/index.php';
-}
-if ( ! function_exists( 'render_block_core_rss' ) ) {
-	require dirname( __FILE__ ) . '/../packages/block-library/src/rss/index.php';
-}
-if ( ! function_exists( 'render_block_core_shortcode' ) ) {
-	require dirname( __FILE__ ) . '/../packages/block-library/src/shortcode/index.php';
-}
-if ( ! function_exists( 'render_block_core_search' ) ) {
-	require dirname( __FILE__ ) . '/../packages/block-library/src/search/index.php';
+/**
+ * Discovers block files from the plugin built artifact, unregistering the
+ * equivalent in core if already defined, then includes the block file.
+ */
+function gutenberg_reregister_core_block_types() {
+	// Blocks directory may not exist if working from a fresh clone.
+	$blocks_dir = dirname( __FILE__ ) . '/../build/block-library/blocks/';
+	if ( ! file_exists( $blocks_dir ) ) {
+		return;
+	}
+
+	$registry = WP_Block_Type_Registry::get_instance();
+
+	$files = scandir( $blocks_dir );
+	foreach ( $files as $file ) {
+		$parts = pathinfo( $file );
+
+		// Ignore all non-PHP files, subdirectories, path traversal.
+		if ( 'php' !== $parts['extension'] ) {
+			continue;
+		}
+
+		// Derive the assumed block name by file path.
+		$block_name = 'core/' . $parts['filename'];
```

Comment:
> It's not always the same, for example:
> https://github.com/WordPress/gutenberg/blob/master/packages/block-library/src/classic/index.js#L13
> 
> Once JSON file is there, it should fix the issue though.

I'm not sure I follow what's meant by "Once JSON file is there", but I think the fact that we can't really rely on the file paths to match the block name could serve as pretty good justification to avoid the newer "discovery"-based approach in favor of an explicit list.

Reply:
I was referring to #13693 and RFC for server side block registration.

It might be better to hardcode it for now.